lib/bing-ads-reporting/service_core.rb

```diff
@@ -1,5 +1,6 @@
 module BingAdsReporting
   class AuthenticationTokenExpired < Exception; end
+  class AuthenticationViaOAuthIsRequired < Exception; end
```

Comment:
I know you followed the same pattern as above but inheriting from `Exception` is a bad thing.

Down the line if something is doing
```
...
rescue => e
```
your `AuthenticationViaOAuthIsRequired` (and `AuthenticationTokenExpired` as well) will never be rescued.

Reply:
So I'd say that unless there is a very specific reason to inherit from `Exception`, we should avoid it.